src/metabase/util/i18n/impl.clj

```diff
@@ -40,7 +49,9 @@
 
   String
   (locale [^String s]
-    (some-> (normalized-locale-string s) LocaleUtils/toLocale))
+    (some-> (normalized-locale-string s)
+            (migrate-legacy-locale-string)
```

Comment:
```suggestion
            migrate-legacy-locale-string
```

The threading macros automatically wrap bare symbol forms after the first arg in parens (for example the `LocalUtils/toLocale` form below)... it's more idiomatic to let it do its thing in cases like this